pgmpy/estimators/base.py

```diff
@@ -121,3 +122,117 @@ def state_counts(self, variable, complete_samples_only=None):
             state_counts = state_count_data.reindex(index=row_index, columns=column_index).fillna(0)
 
         return state_counts
+
+
+class ParameterEstimator(BaseEstimator):
+    def __init__(self, model, data, **kwargs):
+        """
+        Base class for parameter estimators in pgmpy.
+
+        Parameters
+        ----------
+        model: pgmpy.models.BayesianModel or pgmpy.models.MarkovModel or pgmpy.models.NoisyOrModel
+            model for which parameter estimation is to be done
+
+        data: pandas DataFrame object
+            datafame object with column names identical to the variable names of the model.
+            (If some values in the data are missing the data cells should be set to `numpy.NaN`.
+            Note that pandas converts each column containing `numpy.NaN`s to dtype `float`.)
+
+        state_names: dict (optional)
+            A dict indicating, for each variable, the discrete set of states (or values)
+            that the variable can take. If unspecified, the observed values in the data set
+            are taken to be the only possible states.
+
+        complete_samples_only: bool (optional, default `True`)
+            Specifies how to deal with missing data, if present. If set to `True` all rows
+            that contain `np.Nan` somewhere are ignored. If `False` then, for each variable,
+            every row where neither the variable nor its parents are `np.NaN` is used.
+            This sets the behavior of the `state_count`-method.
+        """
+
+        assert set(model.nodes()) <= set(data.columns.values), \
```

Comment:
@chrisittner I think a ```ValueError``` should be raised here. Assertions are more for debugging purposes not for runtime purposes.